src/libraries/System.Memory/tests/System.Memory.Tests.csproj

```diff
@@ -191,6 +191,7 @@
     <Compile Include="ReadOnlySpan\SequenceEqual.long.cs" />
     <Compile Include="ReadOnlySpan\SequenceEqual.T.cs" />
     <Compile Include="ReadOnlySpan\Slice.cs" />
+    <Compile Include="ReadOnlySpan\Split.char.cs" />
```

Comment:
What about T other than char?

Reply:
The PR was still in draft, so was still going to work on more exhaustive tests. What/How many other types do you think would be reasonable to test? Not sure if this is a good idea, but I could generalise the tests so it casts the incoming string of the existing tests to `ReadOnlySpan<byte/long/etc>`. Though of course this means you are realistically testing the same exact data, just with the types differing.

Reply:
> What/How many other types do you think would be reasonable to test?

Just basic validation of an assortment of other types, e.g. byte, int, a custom reference type, and a custom value type.

Reply:
https://gist.github.com/bbartels/720d96d719e2cbf76a3dd8f515dcf196
Is this too ridiculous of a way to go about the generic tests?
Alternatively this? https://gist.github.com/bbartels/9c9df8713e70d50605fa24a98e54952c

Otherwise I'll just duplicate the logic across types like most other tests in `ReadOnlySpanTests`

Reply:
I've added tests going with the alternative route I suggested. The first suggestion above could cut down on the tests a lot. Let me know if you want any changes!